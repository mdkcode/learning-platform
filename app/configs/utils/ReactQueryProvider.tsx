"use client";
import { useState } from "react";
import {
  QueryClientProvider,
  QueryClient,
  QueryCache,
} from "@tanstack/react-query";
import { useErrorHandler } from "@/app/configs/hooks/useErrorHandler";

interface ReactQueryProviderProps {
  children?: React.ReactNode;
}

export const ReactQueryProvider: React.FC<ReactQueryProviderProps> = ({
  children,
}) => {
  const onError = useErrorHandler();
  const [queryClient] = useState(
    () =>
      new QueryClient({
        queryCache: new QueryCache({
          onError,
        }),
      })
  );

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};
