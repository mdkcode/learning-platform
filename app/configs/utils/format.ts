import { Timestamp } from "firebase/firestore";

export function convertTimestampToDateString(created_at: Timestamp) {
  return created_at ? created_at.toDate().toISOString() : "";
}

export function convertSecondsToMinutes(seconds: number) {
  const minutes = Math.ceil(seconds / 60);
  return minutes;
}
