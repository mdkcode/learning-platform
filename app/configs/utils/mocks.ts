export const mockUser = {
  name: "Jane Doe",
  email: "jane.doe@example.com",
  image: "https://example.com/avatar.jpg",
};

export const mockUserSession = {
  user: mockUser,
  expires: new Date().toISOString(),
  accessToken: "dsbhkjscd",
  refreshToken: "894655sdc",
  userId: "1111",
};
