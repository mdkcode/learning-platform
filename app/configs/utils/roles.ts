export enum Roles {
  Student = "STUDENT",
  Admin = "ADMIN",
}
