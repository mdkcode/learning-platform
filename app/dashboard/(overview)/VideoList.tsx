"use client";
import { CourseCard } from "@/app/ui/dashboard/CourseCard";
import { useEffect, useState } from "react";

export interface CourseVideoProps {
  id: number;
  title: string;
  url: string;
  description: string;
}

const VideoList = () => {
  const [videos, setVideos] = useState<CourseVideoProps[]>([]);

  useEffect(() => {
    const fetchVideos = async () => {
      try {
        const response = await fetch("/courseData.json");
        const data = await response.json();
        setVideos(data);
      } catch (error) {
        console.error("Error fetching videos:", error);
      }
    };
    fetchVideos();
  }, []);

  return (
    <div>
      {videos.map((video) => (
        <CourseCard key={video.id} {...video} />
      ))}
    </div>
  );
};

export default VideoList;
