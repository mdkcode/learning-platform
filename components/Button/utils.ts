export enum ButtonType {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  NAVIGATION = "navigation",
}

export const defaultClassName = {
  [ButtonType.PRIMARY]:
    "px-7 py-3 bg-[#00004B] text-white font-semibold rounded-lg shadow-md hover:bg-[#3A59FF] focus:outline-none focus:ring-2 focus:ring-[#3A59FF]",
  [ButtonType.SECONDARY]:
    "px-7 py-3 border-2 border-solid border-[#00004B] bg-white text-[#00004B] font-semibold rounded-lg hover:text-[#3A59FF] hover:border-[#3A59FF]",
  [ButtonType.NAVIGATION]:
    "px-7 py-3 text-white font-semibold rounded-lg bg-gradient-to-r from-orange-400 to-pink-500 hover:bg-gradient-to-r hover:from-orange-500 hover:to-pink-600",
};
