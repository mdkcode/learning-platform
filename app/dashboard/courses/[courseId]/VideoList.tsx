"use client";
import { getCourseVideosById } from "@/app/api/courses/courses.api";
import { convertSecondsToMinutes } from "@/app/configs/utils/format";
import { useQuery } from "@tanstack/react-query";

export const VideoList = ({ id }: { id: string }) => {
  const { data: videos } = useQuery({
    queryKey: [`/dashboard/courses/${id}`, id],
    queryFn: () => getCourseVideosById(id),
    enabled: Boolean(id),
  });

  return (
    <div className="space-y-6">
      <p className="text-2xl font-bold">Course Videos</p>
      {videos?.map(({ id: videoId, title, duration, video_url }) => (
        <div
          key={videoId}
          className="bg-white p-4 rounded-lg shadow-md max-w-3xl sm:max-w-4xl md:max-w-5xl border border-gray-200"
        >
          <div className="flex items-center justify-between mb-4">
            <h3 className="text-xl font-semibold text-gray-800">{title}</h3>
            <p className="text-gray-500">
              {`Duration: Approximately ${convertSecondsToMinutes(
                duration
              )} minutes`}
            </p>
          </div>
          <div className="relative w-full h-0 pb-[56.25%] mb-4">
            <video
              className="absolute top-0 left-0 w-full h-full object-cover rounded-lg"
              src={video_url}
              controls
            />
          </div>
        </div>
      ))}
    </div>
  );
};
