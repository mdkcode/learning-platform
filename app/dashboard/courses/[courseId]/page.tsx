import { NextPageProps } from "@/app/api/general/general.interface";
import { Videos } from "@/app/dashboard/courses/[courseId]/Videos";

export default async function Page({ params }: NextPageProps) {
  const { courseId } = await params;
  return <Videos id={courseId} />;
}
