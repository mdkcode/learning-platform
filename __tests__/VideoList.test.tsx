import { render } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { VideoList } from "@/app/dashboard/courses/[courseId]/VideoList";
import { useQuery } from "@tanstack/react-query";
import { convertSecondsToMinutes } from "@/app/configs/utils/format";

vi.mock("@tanstack/react-query", () => ({
  useQuery: vi.fn(),
}));

vi.mock("@/app/configs/utils/format", () => ({
  convertSecondsToMinutes: vi.fn(),
}));

describe("VideoList", () => {
  it("matches snapshot with videos", () => {
    (useQuery as ReturnType<typeof vi.fn>).mockReturnValue({
      data: [
        {
          id: "vid-1",
          title: "Introduction to the Course",
          duration: 600,
          video_url: "https://example.com/video1.mp4",
        },
        {
          id: "vid-2",
          title: "Deep Dive into React",
          duration: 900,
          video_url: "https://example.com/video2.mp4",
        },
      ],
    });

    (convertSecondsToMinutes as ReturnType<typeof vi.fn>).mockImplementation(
      (seconds: number) => (seconds / 60).toFixed(0)
    );

    const { container } = render(<VideoList id="course-123" />);
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with no videos", () => {
    (useQuery as ReturnType<typeof vi.fn>).mockReturnValue({ data: [] });

    const { container } = render(<VideoList id="course-123" />);
    expect(container).toMatchSnapshot();
  });
});
