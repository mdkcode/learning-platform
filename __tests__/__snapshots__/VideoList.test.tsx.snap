// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VideoList > matches snapshot with no videos 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <p
      class="text-2xl font-bold"
    >
      Course Videos
    </p>
  </div>
</div>
`;

exports[`VideoList > matches snapshot with videos 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <p
      class="text-2xl font-bold"
    >
      Course Videos
    </p>
    <div
      class="bg-white p-4 rounded-lg shadow-md max-w-3xl sm:max-w-4xl md:max-w-5xl border border-gray-200"
    >
      <div
        class="flex items-center justify-between mb-4"
      >
        <h3
          class="text-xl font-semibold text-gray-800"
        >
          Introduction to the Course
        </h3>
        <p
          class="text-gray-500"
        >
          Duration: Approximately 10 minutes
        </p>
      </div>
      <div
        class="relative w-full h-0 pb-[56.25%] mb-4"
      >
        <video
          class="absolute top-0 left-0 w-full h-full object-cover rounded-lg"
          controls=""
          src="https://example.com/video1.mp4"
        />
      </div>
    </div>
    <div
      class="bg-white p-4 rounded-lg shadow-md max-w-3xl sm:max-w-4xl md:max-w-5xl border border-gray-200"
    >
      <div
        class="flex items-center justify-between mb-4"
      >
        <h3
          class="text-xl font-semibold text-gray-800"
        >
          Deep Dive into React
        </h3>
        <p
          class="text-gray-500"
        >
          Duration: Approximately 15 minutes
        </p>
      </div>
      <div
        class="relative w-full h-0 pb-[56.25%] mb-4"
      >
        <video
          class="absolute top-0 left-0 w-full h-full object-cover rounded-lg"
          controls=""
          src="https://example.com/video2.mp4"
        />
      </div>
    </div>
  </div>
</div>
`;
